### ISSUE TITLE
URL Scam Blacklist

### ISSUE THREAD

**Issue by IMXNOOBX** [Vencord/plugin-requests#911]
### Scam Prevention - URL Blacklist & More

As the title suggests, I believe it would be a cool idea to have an open public URL/domain-name blacklist of common scams, maintained by the community. It could be hosted in an open repository, and the plugin would check if any URL detected in the current channel matches one that has already been blacklisted.

I know you explicitly say no API calls, and using an API would be more efficient than downloading the entire domain blacklist into memory, but I'm sure that could be sorted out.

### Things to Take into Account
* I believe most of the Vencord community is well aware of common scam methods, and this would be an extra precaution.
* The plugin could be expanded to verify many other common scam methods, such as:
  * Verifying if a bot named Dyno has the same ID as the original one (to prevent those bot verifications that steal accounts).

### Request Agreement

- [X] I read the above and made sure my plugin is possible and allowed
- [X] I checked the [existing plugins](https://vencord.dev/plugins) and made sure my plugin doesn't already exist
- [X] I [searched the existing issues](https://github.com/Vencord/plugin-requests/issues?q=is%3Aissue) and made sure a similar request doesn't already exist

**Comment by Luna-devv** [comment]
Or just don’t go to weird websites and download everything users tell you to. “Anti Scam” starts at the user and ends at the user. Even if this was a plugin, I’d doubt the people who’d actually benefit from this in some way would even enable the plugin (or even use Vencord)

**Comment by AushevAhmad** [comment]
TL;DR: Best "anti-scam" solution is ✨common sense✨

**Comment by IMXNOOBX** [comment]
I agree with you, but it wouldn't hurt to have some help with it. A simple label next to the link warning that it has been flagged as potentially dangerous could be beneficial. I'm just seeking ideas for a plugin to contribute to this project, so there's no need to be negative about it.

**Comment by Luna-devv** [comment]
It’s more beneficial to just report the site to safe browsing so it’s blocked for every internet user
